test/basic.test.js

```diff
@@ -345,38 +335,33 @@ test('Encapsulation support / 5', t => {
   })
 
   fastify.listen({ port: 0 }, (err, address) => {
-    t.error(err)
+    t.assert.ifError(err)
     sget({
       method: 'GET',
       url: address + '/plugin/nested'
     }, (err, res) => {
-      t.error(err)
-      t.match(res.headers, {
-        'x-middleware-1': 'true',
-        'x-middleware-2': 'true',
-        'x-middleware-3': 'true'
-      })
+      t.assert.ifError(err)
+      t.assert.strictEqual(res.headers['x-middleware-1'], 'true')
+      t.assert.strictEqual(res.headers['x-middleware-2'], 'true')
+      t.assert.strictEqual(res.headers['x-middleware-3'], 'true')
     })
 
     sget({
       method: 'GET',
       url: address + '/plugin'
     }, (err, res) => {
-      t.error(err)
-      t.match(res.headers, {
-        'x-middleware-1': 'true',
-        'x-middleware-2': 'true'
-      })
+      t.assert.ifError(err)
+      t.assert.strictEqual(res.headers['x-middleware-1'], 'true')
+      t.assert.strictEqual(res.headers['x-middleware-2'], 'true')
     })
 
     sget({
       method: 'GET',
       url: address
     }, (err, res) => {
-      t.error(err)
-      t.match(res.headers, {
-        'x-middleware-1': 'true'
-      })
+      t.assert.ifError(err)
+      t.assert.strictEqual(res.headers['x-middleware-1'], 'true')
+      done()
```

Comment:
this `done` could be resolved before the other `sget` execute the `t.assert`.
We should wait for all 3 calls